Makefile

```diff
@@ -0,0 +1,60 @@
+LIBRARY_NAME := $(shell hatch project metadata name 2> /dev/null)
+LIBRARY_VERSION := $(shell hatch version 2> /dev/null)
+
+.PHONY: usage install uninstall check pytest qa build-deps check tag wheel sdist clean dist testdeploy deploy
+usage:
+ifdef LIBRARY_NAME
+	@echo "Library: ${LIBRARY_NAME}"
+	@echo "Version: ${LIBRARY_VERSION}\n"
+else
+	@echo "WARNING: You should 'make dev-deps'\n"
+endif
+	@echo "Usage: make <target>, where target is one of:\n"
+	@echo "install:      install the library locally from source"
+	@echo "uninstall:    uninstall the local library"
+	@echo "dev-deps:     install Python dev dependencies"
+	@echo "check:        perform basic integrity checks on the codebase"
+	@echo "qa:           run linting and package QA"
+	@echo "pytest:       run Python test fixtures"
+	@echo "clean:        clean Python build and dist directories"
+	@echo "build:        build Python distribution files"
+	@echo "testdeploy:   build and upload to test PyPi"
+	@echo "deploy:       build and upload to PyPi"
+	@echo "tag:          tag the repository with the current version\n"
+
+install:
+	python3 -m pip install .
+
+uninstall:
+	python3 -m pip uninstall ${LIBRARY_NAME}
+
+dev-deps:
+	python3 -m pip install -r requirements-dev.txt
+	sudo apt install dos2unix
```

Comment:
typically this kind of thing should NOT be in a makefile. Where is this even used?

Reply:
Say what? You can put whatever you want in a Makefile 🤣

Reply:
Or, to more specifically answer your question, where *do* you put this kind of thing?

This is used in the GitHub workflows to avoid encoding common patterns in GitHub's nonsense yml format. Plus it also makes most of the things you'd want to do during development a `make x` away.

The alternative (and a common CI pattern) is to use a shell script, but that has its own unique set of warts and is obnoxiously painful to invoke on a local machine (`source ci/tools.sh && run_some_ci_action` vs just `make run_some_ci_action`),

This is a pattern I use across all my libraries so I don't have to know or care how to invoke a bunch of arcane commands. Though, granted, `sudo apt install dos2unix` is very Debian-linux specific which is great for GitHub actions but not so great for hypothetical macOS using developers aaannnd I'm totally not writing this from a mac now 👀

Reply:
I would typically put it in you developer setup documentation. I won't block the CL on this